src/main/kotlin/sliv/tool/importer/ProjectParser.kt

```diff
@@ -1,43 +1,116 @@
 package sliv.tool.importer
 
 import javafx.collections.FXCollections
+import javafx.scene.control.Alert
 import javafx.scene.control.TreeItem
+import javafx.scene.image.Image
+import javafx.scene.image.ImageView
+import sliv.tool.parsers.planes.ImagePlanesParser.extractUIDs
+import sliv.tool.project.model.*
+import tornadofx.alert
 import java.io.File
+import kotlin.io.path.Path
 
 object ProjectParser {
-    const val IMAGE_DIRECTORY_NAME = "images"
+    private const val IMAGE_DIRECTORY_NAME = "images"
 
-    val files = FXCollections.observableHashMap<Long, MutableList<OutputInfo>>()
-    val images = FXCollections.observableArrayList<ImageInfo>()
-
-    fun createTreeWithFiles(path: String?, tree: TreeItem<String>): TreeItem<String> {
-        if (path == null) {
-            return tree
+    private fun selectKind(kindString: String): LayerKind? {
+        return when (kindString) {
+            "keypoint" -> LayerKind.KEYPOINT
+            "plane" -> LayerKind.PLANE
+            "line" -> LayerKind.LINE
+            else -> {
+                return null
+            }
         }
-        var directory = File(path)
+    }
+
+    fun parseDirectory(path: String): Project {
+        val directory = File(path)
+        val images = FXCollections.observableHashMap<Long, String>()
+        val landmarks = FXCollections.observableHashMap<Long, MutableList<LandmarkFile>>()
+        val frames = FXCollections.observableArrayList<ProjectFrame>()
+        val layers = FXCollections.observableArrayList<ProjectLayer>()
+        val errorFolders = mutableListOf<String>()
 
-        for (folder in directory.listFiles()) {
-            if (folder.name == IMAGE_DIRECTORY_NAME) {
-                images.addAll(folder.listFiles().map {
-                    ImageInfo(it.nameWithoutExtension, it.absolutePath)
-                })
+        for (folder in directory.listFiles()!!) {
+            if (folder.name != IMAGE_DIRECTORY_NAME) {
+                val directoryName = folder.name
+                val indexOfSeparator = directoryName.lastIndexOf("_")
+                val name = directoryName.substring(0, indexOfSeparator)
```

Comment:
This breaks if directory name doesn't contain '_'. It is better to warn user and skip this directory